src/main/java/com/karankumar/bookproject/ui/book/BookForm.java

```diff
@@ -355,9 +355,12 @@ private boolean isMovedToDifferentShelf() {
     }
 
     private void setBookBean() {
-        Book book = populateBookBean();
-        if (book != null) {
-            binder.setBean(book);
+        // Only repopulate if we create a new Book
+        if (binder.getBean() == null) {
+            Book book = populateBookBean();
+            if (book != null) {
+                binder.setBean(book);
+            }
```

Comment:
Could you please move this into the `else` below? We don't need to check whether the bean in the binder is null twice